File: 119-pascals-triangle-ii/pascals-triangle-ii.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        if(rowIndex == 1) return {1,1};
        vector<int> ans = {1};
        while(rowIndex > 0){
            ans.insert(ans.end(),0);
            ans.insert(ans.begin(),0);
            vector<int> temp = {};
            int i=0,j=1;
            while(j<ans.size()){
                temp.push_back(ans[i]+ans[j]);
                i++;
                j++;
            }
            ans = temp;
            rowIndex--;
        }

        return ans;
    }
};

File: 119-pascals-triangle-ii/pascals-triangle-ii.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        lst = [1]
        if rowIndex == 1:
            return [1,1]
        while rowIndex >0:
            lst[:] = [0] + lst + [0]
            i,j = 0,1
            temp = []
            while j<len(lst):
                temp.append(lst[i]+lst[j])
                i, j = j , j+1
            lst[:] = temp
            rowIndex-=1
        return lst